tests/lib/rules/no-else-return.js

```diff
@@ -182,6 +182,32 @@ ruleTester.run("no-else-return", rule, {
             output: "function foo21() { var x = true; if (x) { return x; } if (x === false) { return false; } }",
             options: [{ allowElseIf: false }],
             errors: [{ messageId: "unexpected", type: "IfStatement" }]
+        },
+        {
+            code: "function foo() { if (true) { return bar; } else { const baz = 1;  return baz; } }",
+            output: null,
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "unexpected", type: "BlockStatement" }]
+        },
+        {
+            code: "function foo() { if (true) { return bar; } else { let baz = 1;  return baz; } }",
+            output: null,
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "unexpected", type: "BlockStatement" }]
+        },
+        {
+            code: "function foo() { let bar = true; if (baz) { return; } else { let bar = false;  return; } }",
+            output: null,
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "unexpected", type: "BlockStatement" }]
+        },
+        {
+            code: "function foo() { let bar = true; if (baz) { return; } else { { let bar = false; }  return; } }",
```

Comment:
This test case could be fixed as well. After removing the `else` and associated braces, `let bar = false` is in its own scope due to the extra braces. So this fix is safe.